### ISSUE TITLE
question about model.py

### ISSUE THREAD

**Issue by Doubaibai** [devsisters/pointer-network-tensorflow#4]
Hi,

I am reading the pointer-network codes. However, I have a question in model.py file. In model.py, line 133-134, you gathered enc_output into dec_inputs. But in the original paper, it seems that the decoder inputs should be from enc_inputs.

133      self.embeded_dec_inputs = tf.stop_gradient(
134      tf.gather_nd(self.enc_outputs, self.idx_pairs))

Could you explain this a little bit?

Best,
Zhe

**Comment by carpedm20** [comment]
Thanks! I've never seen that but you're right. I made a mistake on that and the model should be trained with the ground truth input but the current code does not.

**Comment by Doubaibai** [comment]
Yes. That is what I think. I use the enc_input to train and it seems that the model converged faster. But it is interesting that using enc_output the model also works well.

**Comment by wpf5511** [comment]
I have a question about the stop_gradient op. Using the stop_gradient op, does it mean that the params in  encoder will not changed? Cause the enc_outputs will not changed? Is there something wrong with my understanding?
